data/scripts/talkactions/gm/ban.lua

```diff
@@ -1,47 +1,66 @@
 local ban = TalkAction("/ban")
 
 function ban.onSay(player, words, param)
-	-- create log
 	logCommand(player, words, param)
 
-	local params = param:split(",")
-	if #params < 3 then
-		player:sendCancelMessage("Command requires 3 parameters: /ban <player name>, <duration in days>, <reason>")
+	if param == "" then
+		player:sendCancelMessage("Command param required. Use: /ban playername, days, reason.")
 		return true
 	end
 
-	local playerName = params[1]:trim()
-	local banDuration = tonumber(params[2]:trim())
-	local banReason = params[3]:trim()
+	local name, daysStr, reason = param:match("([^,]+),%s*([^,]+),%s*(.+)")
+	if not name or not daysStr or not reason then
+		player:sendCancelMessage("Invalid command format. Use: /ban playername, days, reason.")
+		return true
+	end
+
+	local banDays = tonumber(daysStr)
+	if not banDays or banDays <= 0 then
+		player:sendCancelMessage("Invalid number of days.")
+		return true
+	end
 
-	if not banDuration or banDuration <= 0 then
-		player:sendCancelMessage("Ban duration must be a positive number.")
+	if banDays > 350000 then
+		player:sendCancelMessage("Ban duration cannot exceed 350000 days.")
 		return true
 	end
 
-	local accountId = Game.getPlayerAccountId(playerName)
+	local accountId = getAccountNumberByPlayerName(name)
```

Comment:
It is interesting to revert this, as the function you used is deprecated, in addition to having the points that Anderson mentioned.